renormalizer/mps/gs.py

```diff
@@ -383,10 +383,10 @@ def eigh_direct(
     nroots = mps.optimize_config.nroots
     if nroots == 1:
         e = w[0]
-        c = v[:, 0]
+        c = v[:, 0] / np.sign(np.max(v[:, 0]))
```

Comment:
I don't think this works as expected. You might first want to take absolute values before calling `max`. Also, this "canonicalize" logic is used below. Maybe it's better to wrap it as a utility function

Reply:
util function added and bug fixed